src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,1051 @@
+    for (std::size_t i = 0; i < this->num_generators(); ++i) {
+      for (std::size_t j = 0; j < this->num_parameters(); ++j) {
+        out[i][j] = multi_filtration_[i][j];
+      }
+    }
+    return out;
+  }
+
+  inline void _clean(bool keep_inf=true) {
+    multi_filtration_.erase(std::remove_if(multi_filtration_.begin(), multi_filtration_.end(),
+                              [keep_inf](const OneCritical &a) {
+                                return a.empty() ||
+                                       ((!keep_inf) &&
+                                        (a.is_inf() || a.is_minus_inf()));
+                              }),
+                            multi_filtration_.end());
+  }
+  inline std::size_t num_generators() const {
+    return multi_filtration_.size();
+  }
+
+
+
+  // TODO : this costs a lot... optimize / cheat in some way for python ?
+  /*
+  * Checks if `this`, seen as a birth curve is under the `other` birth curve,
+  *
+  */
+  inline bool operator<(const Multi_critical_filtration<T,co> &other) const {
+    //check if this curves is below other's curve 
+    // ie for each guy in this, check if there is a guy in other that dominates him
+    for (std::size_t i = 0u; i < multi_filtration_.size(); ++i) {
+      for (std::size_t j = 0u; j < other.multi_filtration_.size(); ++j) {
+        // i<j
+        if (strictly_dominates(other.multi_filtration_[j],multi_filtration_[i]))
+          continue;
+      }
+      return false;
+    }
+    return true;
+  }
+  /*
+  * Checks if `this`, seen as a birth curve is over the `other` birth curve,
+  */
+  inline bool operator>(const Multi_critical_filtration<T,co> &other) const {
+    return other < *this;
+  }
+
+  /*
+  * Checks if `this`, seen as a birth curve is under the `other` birth curve,
+  */
+  inline bool operator<=(const Multi_critical_filtration<T,co> &other) const {
+    //check if this curves is below other's curve 
+    // ie for each guy in this, check if there is a guy in other that dominates him
+    for (std::size_t i = 0u; i < multi_filtration_.size(); ++i) {
+      for (std::size_t j = 0u; j < other.multi_filtration_.size(); ++j) {
+        // i <= j 
+        if (dominates(other.multi_filtration_[j], multi_filtration_[i]))
+          continue;
+      }
+      return false;
+    }
+    return true;
+  }
+  /*
+  * Checks if `this`, seen as a birth curve is over the `other` birth curve,
+  */
+  inline bool operator>=(const Multi_critical_filtration<T,co> &other) const {
+    return other <= *this;
+  }
+
+public:
+  // for compiler
+  constexpr static const T T_inf = One_critical_filtration<T>::T_inf;
+  constexpr static const bool is_multi_critical = true;
+
+private:
+  std::vector<One_critical_filtration<T>> multi_filtration_;
+};
+
+} // namespace Gudhi::multiparameter::multi_filtrations
+
+namespace std {
+
+template <typename T>
+class numeric_limits<Gudhi::multiparameter::multi_filtrations::
+                         One_critical_filtration<T>> {
```

Comment:
Where do you need it (trying to understand if it is also needed for multi-critical)?